transforms_crud.yml#data_frame/transforms_crud/Test basic transform crud failure on CI

One of my PR builds hit this: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+pull-request-2/10073/console

Does not reproduce locally with:
```
./gradlew :x-pack:plugin:integTestRunner -Dtests.seed=FDB44C20FECF6CB5 -Dtests.class=org.elasticsearch.xpack.test.rest.XPackRestIT -Dtests.method="test {p0=data_frame/transforms_crud/Test basic transform crud}" -Dtests.security.manager=true -Dtests.locale=ar-EG -Dtests.timezone=Etc/GMT-3 -Dcompiler.java=11 -Druntime.java=8
```

Failure info & Stack trace:
```
  1> [2019-03-15T21:19:58,951][INFO ][o.e.x.t.r.XPackRestIT    ] [test] Stash dump on test failure [{
  1>   "stash" : {
  1>     "body" : {
  1>       "count" : 2,
  1>       "transforms" : [
  1>         {
  1>           "id" : "airline-transform-dos",
  1>           "source" : "airline-data",
  1>           "dest" : "airline-data-by-airline-again",
  1>           "query" : {
  1>             "match_all" : { }
  1>           },
  1>           "pivot" : {
  1>             "group_by" : {
  1>               "airline" : {
  1>                 "terms" : {
  1>                   "field" : "airline"
  1>                 }
  1>               }
  1>             },
  1>             "aggregations" : {
  1>               "avg_response" : {
  1>                 "avg" : {
  1>                   "field" : "responsetime"
  1>                 }
  1>               }
  1>             }
  1>           }
  1>         },
  1>         {
  1>           "id" : "airline-transform",
  1>           "source" : "airline-data",
  1>           "dest" : "airline-data-by-airline",
  1>           "query" : {
  1>             "match_all" : { }
  1>           },
  1>           "pivot" : {
  1>             "group_by" : {
  1>               "airline" : {
  1>                 "terms" : {
  1>                   "field" : "airline"
  1>                 }
  1>               }
  1>             },
  1>             "aggregations" : {
  1>               "avg_response" : {
  1>                 "avg" : {
  1>                   "field" : "responsetime"
  1>                 }
  1>               }
  1>             }
  1>           }
  1>         }
  1>       ]
  1>     }
  1>   }
  1> }]
  1> [2019-03-15T21:19:59,087][INFO ][o.e.x.t.r.XPackRestIT    ] [test] [p0=data_frame/transforms_crud/Test basic transform crud] after test
FAILURE 0.47s | XPackRestIT.test {p0=data_frame/transforms_crud/Test basic transform crud} <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: Failure at [data_frame/transforms_crud:99]: transforms.0.id didn't match expected value:
   >                transforms.0.id: expected [airline-transform] but was [airline-transform-dos]
   > 	at __randomizedtesting.SeedInfo.seed([FDB44C20FECF6CB5:75E073FA5033014D]:0)
   > 	at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:393)
   > 	at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:370)
   > 	at java.lang.Thread.run(Thread.java:748)
   > Caused by: java.lang.AssertionError: transforms.0.id didn't match expected value:
   >                transforms.0.id: expected [airline-transform] but was [airline-transform-dos]
   > 	at org.elasticsearch.test.rest.yaml.section.MatchAssertion.doAssert(MatchAssertion.java:93)
   > 	at org.elasticsearch.test.rest.yaml.section.Assertion.execute(Assertion.java:76)
   > 	at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:386)
   > 	... 37 more
```

[Full build log saved for posterity](https://github.com/elastic/elasticsearch/files/2972641/consoleText.txt)

--- a/x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/persistence/DataFrameTransformsConfigManager.java
+++ b/x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/persistence/DataFrameTransformsConfigManager.java
@@ -39,6 +39,7 @@ import org.elasticsearch.index.engine.VersionConflictEngineException;
 import org.elasticsearch.index.query.BoolQueryBuilder;
 import org.elasticsearch.index.query.QueryBuilders;
 import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.search.sort.SortOrder;
 import org.elasticsearch.xpack.core.ClientHelper;
 import org.elasticsearch.xpack.core.dataframe.DataFrameField;
 import org.elasticsearch.xpack.core.dataframe.DataFrameMessages;
@@ -141,6 +142,7 @@ public class DataFrameTransformsConfigManager {
         SearchRequest request = client.prepareSearch(DataFrameInternalIndex.INDEX_NAME)
             .setIndicesOptions(IndicesOptions.LENIENT_EXPAND_OPEN)
             .setTrackTotalHits(true)
+            .addSort(DataFrameField.ID.getPreferredName(), SortOrder.ASC)
             .setSize(DEFAULT_SIZE)
             .setQuery(queryBuilder)
             .request();

